apiserver/facades/controller/usersecretsdrain/drain.go

```diff
@@ -0,0 +1,199 @@
+// Copyright 2023 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package usersecretsdrain
+
+import (
+	"github.com/juju/errors"
+	"github.com/juju/loggo"
+
+	commonsecrets "github.com/juju/juju/apiserver/common/secrets"
+	apiservererrors "github.com/juju/juju/apiserver/errors"
+	coresecrets "github.com/juju/juju/core/secrets"
+	"github.com/juju/juju/rpc/params"
+	"github.com/juju/juju/secrets"
+	"github.com/juju/juju/secrets/provider"
+)
+
+var logger = loggo.GetLogger("juju.apiserver.usersecretsdrain")
+
+// SecretsDrainAPI is the implementation for the SecretsDrain facade.
+type SecretsDrainAPI struct {
+	*commonsecrets.SecretsDrainAPI
```

Comment:
We don't want to embed APIs any more. It makes versioning of embedding APIs cumbersome. Make it a member and call through to it.